PERL/ASGSUtil.pm

```diff
@@ -68,6 +68,18 @@ sub writeJSON {
    }
    # add the name of this script
    $jshash_ref->{"self"} = $calling_script;
+   return;
+}
+#
+# write out the JSON file containing additional parameters
+# as added by this script
+sub writeJSON {
+   # json passed by reference and modified in this subroutine
+   my ( $jshash_ref ) = @_;
+   my ( $package, $filename, $line ) = caller;
+   my $calling_script = File::Basename::basename($filename);
+   # add the name of this script
+   $jshash_ref->{"self"} = $calling_script;
```

Comment:
hash key barewords don't need quotes, double quotes also are different than single since they imply some auto-expansion of varies so there's a slight overhead with the checking of the contents. Not work fixing here, just a note.

Reply:
Thank you, these are subtle differences that I was not sure about. I will use barewords in the future.